components/Player/NotesTable.tsx

```diff
@@ -1,24 +1,144 @@
+}> = ({ note }) => {
+  const [edit, setEdit] = useState(false);
+  const [currNote, setCurrNote] = useState<Notes>(note);
+  const router = useRouter();
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const editNote = () => {
+    setCurrNote(note);
+    setEdit(true);
+  };
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const toasty = () => toast.success("Note deleted!");
+  const refreshProfile = useCallback(() => {
+    router.replace(router.asPath);
+  }, [router]);
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  async function deleteNote(): Promise<void> {
+    const response = await fetch(`/api/notes/delete?id=${note.id}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (!response.ok) {
+      throw await response.json();
+    }
+    refreshProfile();
+    toasty();
+  }
+  return (
+    <>
+      <Menu>
+        {({ open }) => (
+          <>
+            <Menu.Button className="relative focus:outline-none flex align-center justify-center">
+              <Icon type="more" className="h-5 ml-4 fill-current" />
+            </Menu.Button>
+            <Transition
+              show={open}
+              enter="transition duration-100 ease-out"
+              enterFrom="transform scale-95 opacity-0"
+              enterTo="transform scale-100 opacity-100"
+              leave="transition duration-75 ease-out"
+              leaveFrom="transform scale-100 opacity-100"
+              leaveTo="transform scale-95 opacity-0"
+            >
+              <Menu.Items
+                className="absolute z-10 border-medium-gray shadow-lg bg-white rounded-md pt-12 focus:outline-none flex flex-col text-unselected font-semibold text-sm w-32"
+                style={{ borderWidth: 1, transform: "translateY(-32px)" }}
+                static
+              >
+                <Icon type="more" className="h-5 ml-4 -mt-10 fill-current" />
+                <Menu.Item>
+                  {({ active }) => (
+                    <button
+                      type="button"
+                      className={`flex items-center w-full px-4 py-2 font-medium ${
+                        active ? "bg-button text-dark" : ""
+                      }`}
+                      onClick={editNote}
+                    >
+                      <Icon type="edit" className="h-4 mr-3 fill-current" />
+                      <p className="justify-self-start">Edit</p>
+                    </button>
+                  )}
+                </Menu.Item>
+                <Menu.Item>
+                  {({ active }) => (
+                    <button
+                      type="button"
+                      className={`flex items-center w-full px-4 py-2 rounded-b-md font-medium ${
+                        active ? "bg-button text-dark" : ""
+                      }`}
+                      onClick={deleteNote}
```

Comment:
Let's add a quick modal for delete to confirm that we want this note to be deleted! It will look very similar to all our other delete confirmation modals!